enoki/matrix.py

```diff
@@ -267,30 +267,46 @@ def matrix_to_quat(m):
 
 
 def quat_to_euler(q):
-    q_y_2 = _ek.sqr(q.y)
+    name = _ek.detail.array_name('Array', q.Type, [3], q.IsScalar)
+    module = _modules.get(q.__module__)
+    Array3f = getattr(module, name)
 
+    sinp = 2 * _ek.fmsub(q.w, q.y, q.z * q.x)
+    gimbal_lock = _ek.abs(sinp) > 1.0 - 1e-5
```

Comment:
```suggestion
    gimbal_lock = _ek.abs(sinp) > (1.0 - 1e-5)
```
Just to clarify your intentions 😄

Reply:
Where does `1e-5` come from? Could you use `ek.Epsilon(type(sinp))` instead?

Reply:
Or even `_ek.OneMinusEpsilon(type(sinp))`